backend/substrapp/compute_tasks/image_builder.py

```diff
@@ -42,17 +42,18 @@ def wait_for_image_built(function: orchestrator.Function, channel: str) -> None:
 def load_remote_function_image(function: orchestrator.Function, channel: str) -> None:
     container_image_tag = utils.container_image_tag_from_function(function)
     # Ask the backend owner of the function if it's available
-    logger.info(
-        f"Initial function URI {function.function_address.uri}; "
-        f"modified URI{function.function_address.uri.replace('file', 'image')}"
-    )
+
+    # Necessary to query the DB again to get an updated Function object
+    # Else the image URI is not available
+    api_function = ApiFunction.objects.get(key=function.key)
 
     function_image_content = organization_client.get(
         channel=channel,
         organization_id=function.owner,
-        # TODO create a clean Address for function image
-        url=function.function_address.uri.replace("file", "image"),
-        checksum=None,
+        url=function.image.uri,
+        checksum=function.image.checksum,
+        # url=api_function.image_address,
+        # checksum=api_function.image_checksum,
```

Comment:
Some commented code to delete. Also, the comment suggests that we should use `api_function` isntead of `function` but the `.get` do otherwise

Reply:
Back to api function in the end, for the reasons indicated in the comment